src/main/java/TaskList.java

```diff
@@ -0,0 +1,158 @@
+
+        switch (taskType) {
+        case "todo":
+            addTask(new Task(commandParts[1]));
+            break;
+        case "deadline":
+            String[] deadlineParts = commandParts[1].split("/by" , 2);
+            if (deadlineParts.length != 2) {
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                addTask(new Deadline(deadlineParts[0], deadlineParts[1]));
+            }
+            break;
+        case "event":
+            String[] eventparts = commandParts[1].split("/from" , 2);
+            if (eventparts.length != 2) {
+                System.out.println("Invalid deadline format! Use: deadline <<description>> /by <<deadline>>.");
+            } else {
+                String[] endDateParts = eventparts[1].split("/to" , 2);
+                if (endDateParts.length != 2) {
+                    System.out.println("☹ OOPS!!! Invalid event format. Use 'event <description> /from <start date/time> /to <end date/time>'.");
+                } else {
+                    addTask(new Events(eventparts[0], endDateParts[0], endDateParts[1]));
+                }
+            }
+            break;
+        case "mark":
+            if (commandParts.length == 1 || !isValidIndex(Integer.parseInt(commandParts[1])-1) ) {
+                System.out.println("Invalid command format for marking a task. Use 'mark <index>'.");
+            } else {
+                markTask(Integer.parseInt(commandParts[1]) - 1);
+            }
+            break;
+        case "unmark":
+            if (commandParts.length == 1 || !isValidIndex(Integer.parseInt(commandParts[1])-1) ) {
+                System.out.println("Invalid command format for marking a task. Use 'mark <index>'.");
+            } else {
+                unmarkTask(Integer.parseInt(commandParts[1]) - 1);
+            }
+            break;
+        default:
+            System.out.println("☹ Dobby does not understand."); // Default to addTask if not recognised
+            break;
+        }
+    }
+
+        /**
+         * Adds user input into taskList
+         *
+         * @param task input given by user
+         */
+        public void addTask(Task task) {
+            taskList[taskCount] = task;
+            //isMarked[taskCount] = false;
+            taskCount += 1;
+            System.out.println("added: " + task);
+        }
+
+        /**
+         * list out current tasks and displays task status
+         */
+        public void listTasks() {
+            if (taskCount == 0) {
+                System.out.println("Dobby has no tasks :(");
+                return;
+            }
+            System.out.println("List\n~~~~~~~~~~~~~~~~");
+            for (int i = 0 ; i < taskCount ; i += 1) {
+                System.out.println("  " + (i+1) + ". [" + taskList[i].getType() + "]" + taskList[i]);
+            }
+            System.out.println("~~~~~~~~~~~~~~~~");
```

Comment:
In general, you could consider avoiding some reused string literals (not restricted to this method and file) by refactoring to a named constant to adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)